sklearn/pipeline.py

```diff
@@ -869,6 +874,9 @@ def _validate_transformers(self):
                 raise TypeError("All estimators should implement fit and "
                                 "transform. '%s' (type %s) doesn't" %
                                 (t, type(t)))
+            elif not hasattr(t, "transform"):
+                warnings.warn("Intermediate step '%s' (type %s) does not have "
```

Comment:
Intermediate step -> transformer

This change isn't tested. 

I'm also happy to see it removed

Reply:
Do you mean removing the entire warning altogether?  I thought that `FeatureUnion` should be similar in behaviour to Pipeline.

Reply:
It probably should behave similar to Pipeline, but it's somewhat less valuable where `transform` is not supported! I'd also be happy to pragmatically merge functionality for Pipeline and deal with FeatureUnion later in a separate pull request.

Reply:
Alright, I have removed it for now.